cli/server/server.go

```diff
@@ -259,17 +266,28 @@ func restoreDB(ctx *cli.Context) error {
 	defer prometheus.ShutDown()
 	defer pprof.ShutDown()
 
-	var allBlocks = reader.ReadU32LE()
+	start := uint32(0)
+	dumpSize := reader.ReadU32LE()
+	if ctx.Bool("diff") {
+		// in diff first uint32 is the index of the first block
+		start = dumpSize
+		dumpSize = reader.ReadU32LE()
+	}
 	if reader.Err != nil {
 		return cli.NewExitError(err, 1)
 	}
-	if skip+count > allBlocks {
-		return cli.NewExitError(fmt.Errorf("input file has only %d blocks, can't read %d starting from %d", allBlocks, count, skip), 1)
+	if skip < start {
+		return cli.NewExitError(fmt.Errorf("input file start from %d block, can't skip %d", start, skip), 1)
```

Comment:
I'm not sure that's semantically correct. Skip now has a meaning of "number of block blobs to skip" and not of "import start height".

Reply:
Well, maybe rename the flag then? I find it rather inconvenient to provide `skip` value, because it is a difference between 2 random numbers (#blocks in store and #first block in dump) and user doesn't want to do any math.

Reply:
OK, let's do
```
                        Name:  "start, s",
                        Usage: "block number to start from (default: 0)",
```
then. It's gonna be the first one to import, probably the most convenient form (like I know I'm working on a block 2400907, so that's the one I'm specifying here). But we should also note in documentation that block 0 is actually NEVER imported, it's just checked to be the same as the one generated locally for the chain configured.

It'll make `-s 2400907` compatible with the previous meaning of `--skip` because we had to skip N blocks (including block 0) to import block with index N and it'll also be more appropriate for diff dumps.

Reply:
Fixed.